packages/repository-json-schema/src/__tests__/integration/build-schema.integration.ts

```diff
@@ -552,6 +552,54 @@ describe('build-schema', () => {
       });
     });
 
+    context('model conversion', () => {
+      @model()
+      class Category {
+        @property.array(() => Product)
+        products?: Product[];
+      }
+
+      @model()
+      class Product {
+        @property(() => Category)
+        category?: Category;
+      }
+
+      const expectedSchema = {
+        title: 'Category',
+        properties: {
+          products: {
+            type: 'array',
+            items: {$ref: '#/definitions/Product'},
+          },
+        },
+        definitions: {
+          Product: {
+            title: 'Product',
+            properties: {
+              category: {
+                $ref: '#/definitions/Category',
+              },
+            },
+          },
+          Category: {
+            title: 'Category',
+            properties: {
+              products: {
+                type: 'array',
+                items: {$ref: '#/definitions/Product'},
+              },
+            },
+          },
+        },
+      };
+
+      it('handles circular references', () => {
+        const schema = modelToJsonSchema(Category);
```

Comment:
Please add a test to verify `getJsonSchema` too.

Reply:
WIP, will add it in the next commit.

Reply:
I didn't realize `getJsonSchema` is the entry point of the schema generation, thought it starts with `modelToJsonSchema`.

Good catch! Tests added.